### PR TITLE
Upgrade mdox with cache and some http settings to reduce CI failures

### PR DESCRIPTION
**Pull request by douglascamata**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

- `mdox` as upgraded to the most recent commit in the `master` branch: https://github.com/bwplotka/mdox/commit/25b9abcf90a064d41422f45900b389f50c1b8ec2. This head includes:
  - Cache of the link checks (added in https://github.com/bwplotka/mdox/commit/25b9abcf90a064d41422f45900b389f50c1b8ec2).
  - Important configurations for link checks (added in https://github.com/bwplotka/mdox/commit/d39d57f434dd89a279ff267c96535a8f7eedb715, see the [docs](https://github.com/bwplotka/mdox#link-validation-configuration)). For the moment we bump the max connections per host and set the timeout of each request to 1 minute. Other changes can be done later as we see how the CI behaves.
- Added a Github Actions cache for the `mdox` cache file based on the hash of the markdown files it works with.

## Verification

* Ran `make check-docs` locally.
* Verified that cache builds and restores worked fine.

### CODE REVIEW

File: .mdox.validate.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 version: 1
+timeout: '1m'
+host_max_conns: 10
+cache:
+  type: 'SQLite'
```
Comment (saswatamcode) [inline .mdox.validate.yaml:5]: Maybe adding a `jitter` param with a duration of a few days would be good too? 🙂

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -22,10 +22,19 @@ jobs:
...
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - uses: actions/cache@v3
+        with:
+          path: .mdoxcache
+          key: ${{ runner.os }}-mdox-${{ hashFiles('docs/**/*.md', 'examples/**/*.md', 'mixin/**/*.md', '*.md') }}
+          restore-keys: |
+            ${{ runner.os }}-mdox-
```
Comment (saswatamcode) [inline .github/workflows/docs.yaml:37]: I wonder if a maintainer could check if this is actually in cache now with https://docs.github.com/en/rest/actions/cache#list-github-actions-caches-for-a-repository. Would be neat to check beforehand!

**Review by saswatamcode — DISMISSED** [review]
Thanks a lot! This should make docs check way faster and less flaky! 💪🏻 

Some tiny suggestions!

File: .mdox.validate.yaml (hunk 1 above)
Comment (douglascamata) [reply .mdox.validate.yaml:5]: Oh, my bad. I thought there was a default jitter. Shall we use `validity/2` as jitter? What do you think?

Comment (saswatamcode) [reply .mdox.validate.yaml:5]: SGTM!

Comment (douglascamata) [reply .mdox.validate.yaml]: Done in https://github.com/thanos-io/thanos/pull/5500/commits/7ec0faecec4827973a88db63ea31c549a25d8833

**Pull request merged**